core/com.b2international.snowowl.datastore.server/src/com/b2international/snowowl/datastore/server/DatastoreServerBootstrap.java

```diff
@@ -147,9 +150,26 @@ public void run(SnowOwlConfiguration configuration, Environment env, IProgressMo
 			initializeJobSupport(env, configuration);
 			initializeRepositories(configuration, env);
 			initializeRequestSupport(env, configuration.getModuleConfig(RepositoryConfiguration.class).getNumberOfWorkers());
+			verifyRepositories(env);
 		}
 	}
 	
+	@SuppressWarnings("deprecation")
+	private void verifyRepositories(Environment env) {
+		final DefaultRepositoryManager repositories = (DefaultRepositoryManager) env.service(RepositoryManager.class);
+		Optional<InternalRepository> inConsistentRepository = repositories.repositories()
+			.stream()
+			.filter(repository -> repository instanceof InternalRepository).map(InternalRepository.class::cast)
+			.filter(repository -> !repository.isConsistent(BranchPathUtils.createMainPath())) 
+			/* XXX: or should we check all the branches, not just the main??*/
+			.findAny();
+		
+		if (inConsistentRepository.isPresent()) {
+				LOG.error("Found inconsistent repository: {}, halting boot.", inConsistentRepository.get().getCdoRepository().getRepositoryName());
+				Thread.currentThread().suspend();
```

Comment:
Suspending the thread is dangerous; while it certainly blocks further operations here, we don't know what's running on the side, and it should still be possible to shut down cleanly as well.

Existing examples that break the startup mid-process are throwing `com.b2international.snowowl.core.SnowOwlApplication.InitializationException` if a problem is encountered. Can you check if it would work here as well?